### DISCUSSION TITLE
[Enhancement] Support DependencyObject.SetCurrentValue

### DISCUSSION THREAD

**Discussion by andyb1979** [dotnet/maui#176] [General]
## Summary
DependencyObject.SetCurrentValue() is available in WPF platform but missing in UWP 

## API Changes

[DependencyObject.SetCurrentValue](https://docs.microsoft.com/en-us/dotnet/api/system.windows.dependencyobject.setcurrentvalue?view=netcore-3.1) sets the value of a DependencyProperty without changing its value source. 

Setting a DependencyProperty value in code obeys Dependency Property Precedence, but also any bindings to that property, template bindings or style setters to that property are overwritten, since setting a DProp in code has the 2nd highest precedence. 

DependencyObject.SetCurrentValue allows you to set the value in code while propagating the value back to binding sources with TwoWay binding, and without overwriting the bindings or style setters. 

## Intended Use Case

Imagine you're a custom control vendor [like us](https://www.scichart.com/). We supply a control that has a property 

```
public class MyCoolControl : UserControl 
{
     public static readonly DependencyProperty SomethingProperty= DependencyProperty.Register("Something", typeof(double), typeof(MyCoolControl), new PropertyMetadata(null));


    public double Something 
    {
        get => (double)GetValue(SomethingProperty);
        set => SetValue(SomethingProperty, value);
    }
}
```

We want our users to bind to this property and control it in a ViewModel

```
<local:MyCoolControl Something="{Binding SomethingInViewModel, Mode=TwoWay}"/>
```

So far so good. But what if we had some logic inside our control that also needed to update the same property and propagate those changes to the ViewModel? 

```
public class MyCoolControl : UserControl 
{
     public static readonly DependencyProperty SomethingProperty= DependencyProperty.Register("Something", typeof(double), typeof(MyCoolControl), new PropertyMetadata(null));


    public double Something 
    {
        get => (double)GetValue(SomethingProperty);
        set => SetValue(SomethingProperty, value);
    }


    private void InternalLogic() 
    {
        Something = 123.4; // Erases bindings on this property!!! 
        this.SetCurrentValue(SomethingProperty, 123.4); // This works fine, propagating 123.4 to the ViewModel
    }
}
```

## Concrete Example

We have a concrete example in our [WPF Chart Control - SciChart](https://www.scichart.com/wpf-chart-features/). 

- [Axis.VisibleRange](https://www.scichart.com/documentation/win/current/Axis%20Ranging%20-%20AutoRange%20and%20VisibleRange.html) is a DependencyProperty
- Users bind Axis.VisibleRange to a ViewModel TwoWay
- However Zooming and Panning by mouse-click on our parent control must update Axis.VisibleRange. We want those changes to propagate to user viewmodel and not to erase user bindings.

**Comment by Monasha1992** [comment]
Maybe [Attached properties](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/attached-properties) could help you.
